### PR TITLE
feat: CPLYTM-1302 - include a constitution file for speckit

### PR DESCRIPTION
**Pull request by marcusburghardt**
## Summary

We want to make the collaboration more efficient by using Speckit and AI tools.
This PR introduces the SpecKit constitution file, which includes the core principles used by SpecKit commands.

## Related Issues

- Standardize SpecKit core principles.

## Review Hints

- It was initially based on [project style guides](https://github.com/complytime/community/blob/main/STYLE_GUIDE.md)
- A very similar version was already tested in https://github.com/complytime/complyctl/pull/381
- This version contain minor wording improvements adapted to code agents.

### CODE REVIEW

File: .specify/memory/constitution.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+- **Self-Documenting**: The code structure itself MUST explain the logic. Comments MUST explain the *why* (business logic/intent), not the *what* (syntax).
+
+**Rationale**: Readable code reduces onboarding time, prevents bugs from misunderstanding, and enables faster debugging.
+
+### V. Do Not Reinvent the Wheel
+
+Leverage existing solutions, but validate their quality and contribute back.
+
+**Implementation**:
+- **Research First**: Always research existing open-source libraries or cloud-native solutions before writing custom code.
```
Comment (jpower432) [inline .specify/memory/constitution.md]: I think here we should actually ask the agent to prompt the user for a dependency list. Not the the evaluation umprompted. WDYT?

File: .specify/memory/constitution.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+- **Vet Dependencies**: Evaluate the library's reliability: check its adoption rate, governance model, maintenance frequency (last commit date), and community health.
+- **Contribute Upstream**: If an existing library is close to what we need but missing a feature or bug fix, propose a separate PR contributing to it. Prefer sending a Pull Request to the upstream repository over creating a local workaround or a hard fork.
```
Comment (jpower432) [inline .specify/memory/constitution.md]: This is more human oriented

File: .specify/memory/constitution.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+## Coding Standards
+
+### Guidelines for All Programming Languages
+
+- **Empty Line at End of File**: Ensure that all files include an empty line at the end. This helps with version control diffs and adheres to POSIX standards.
+- **Pre-commit Hooks**: The pre-commit and pre-push hooks can be configured by installing [pre-commit](https://pre-commit.com/).
+- **Makefile**: Use Makefile to centralize code specific commands.
+- **Testing**: Write tests for your code. Use descriptive names for test functions and include edge cases. Always test inputs from external sources. Ensure at least a positive and a negative test case to ensure errors and exceptions covered and properly treated.
+- **Line Length**: Limit lines to 99 characters unless in exceptional cases where it is reasonable to improve readability.
+- **Lint**: Ensure no lint issues according to the lint settings defined in the repository. No trailing spaces.
```
Comment (jpower432) [inline .specify/memory/constitution.md]: Should we tell it where the lint setting are?

**Review by jpower432 — COMMENTED** [review]
A couple comments here

File: .specify/memory/constitution.md (hunk 3 above)
Comment (marcusburghardt) [reply .specify/memory/constitution.md]: Good idea.

File: .specify/memory/constitution.md (hunk 1 above)
Comment (marcusburghardt) [reply .specify/memory/constitution.md]: @jpower432 , I used AI to support me with complete review and incorporated your feedback in the last commit. Could you take a look, please?

File: .specify/memory/constitution.md (hunk 2 above)
Comment (marcusburghardt) [reply .specify/memory/constitution.md]: Reviewed the whole file again and consistently reworded.

File: .specify/memory/constitution.md (hunk 3 above)
Comment (marcusburghardt) [reply .specify/memory/constitution.md]: Implemented in the last commit.

**Comment by marcusburghardt** [comment]
Closing since the commits are now incorporated into https://github.com/complytime/complyctl/pull/381
